R/linelist_merge.R

```diff
@@ -0,0 +1,416 @@
+    names(og_sheet) <-gsub(x= names(og_sheet),pattern ="\\s+$", replacement ="")
+    names(og_sheet) <-gsub(x= names(og_sheet),pattern ="\\s+|[[:punct:]]", replacement ="_")
+    names(og_sheet) <-tolower(names(og_sheet))
+    names(og_sheet)<- stringi::stri_trans_general(names(og_sheet) , "Latin-ASCII")
+
+    #remove all accents from values in dataframe
+    #replace all values in file that should be coded as missing
+    #using gsub and regex pattern matching here as some patterns require to be fixed and others not
+    og_sheet<-og_sheet %>% mutate_all(stringi::stri_trans_general,"Latin-ASCII") %>%
+      mutate_all(gsub,pattern="['?]",replacement = "",ignore.case = T, perl = T) %>%
+      mutate_all(gsub, pattern="(?i)^NA$|(?i)^N/A$|(?i)^N/A,|(?i)^N\\A$|(?i)^Unknown$|(?i)^dont know$|(?i)^Unkown$|(?i)^N.A$|(?i)^NE SAIT PAS$|(?i)^inconnu$|^ $|(?i)^Nao aplicavel$|(?i)^Sem informacao$", replacement=NA, perl = T)
+#must keep this pattern all one line or doesnt work
+
+    # filter variable cleaning dictionary specific to country file loaded
+    var_dict_country<- var_dict %>% filter(country==iso)
+    var_dict_country<- var_dict_country[var_dict_country$old_var_r %in% names(og_sheet),]
+
+    # list variables to drop that are non-template
+    drop_nontemplate_vars<- as.list(var_dict_country$old_var_r[var_dict_country$notes=="non_template" & !is.na(var_dict_country$notes)])
+
+    # drop variables from dataframe which are non-tmeplate
+    if(length(drop_nontemplate_vars)!=0){
+      og_sheet<-og_sheet[ , -which(names(og_sheet) %in% drop_nontemplate_vars)]
+    }
+
+    # if file requires no cleaning as matched WHO standard template do this loop
+    if(template==TRUE){
+      output_sheet<-og_sheet
+
+     #If file needs recoding of variables/merging of variables
+      }else if(template==FALSE){
+      # recode variables
+      recode_vars<-as.list(var_dict_country$old_var_r[grepl("Recode",var_dict_country$notes) & !is.na(var_dict_country$notes)])
+
+
+      if(length(recode_vars)!=0){
+      var_dict_country<- var_dict_country[var_dict_country$old_var_r %in% names(og_sheet),]
+      #from variable dictionary find which variables require recode
+      #these variables are usually patinfo_occus (proffession), comcond_exist (pre exsiting medical) and patsympt_other (other symptoms whihc have often been split inot separate variables)
+      variable_to<- sub("Recode to ", "",as.list(var_dict_country$notes[grepl("Recode",var_dict_country$notes) & !is.na(var_dict_country$notes)]))
+      what_to_match<-unique(variable_to)
+
+      recode_sheet<- og_sheet[ , which(names(og_sheet) %in% recode_vars)]
+
+      #remove following instances from variable names taht need recoding for the next step to leave what will become the recoded value
+      names(recode_sheet)<- gsub(x = names(recode_sheet), pattern = "COVID.|Co19.|comcond.|patsympt.|patinfo.", replacement = "", ignore.case = T)
+
+      #replace all instances of no with missing
```

Comment:
you could save yourself some density here by using a grepl for the Yes options and then if sum is the number you want

Reply:
TRUE/FALSE is treated as 1/0

Reply:
some columns that are part of the recode have values that I want to keep still (that are not yes), its an annoying step but I think is only safe way round it